express-notes-api/server.js

```diff
@@ -0,0 +1,184 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+/* This just makes it easier to create error messages for me.
+   And it looks cleaner when written in the code later on. */
+class Error {
+  constructor(errorMsg) {
+    this.error = errorMsg;
+  }
+}
+
+const port = 8080;
+const readFile = './data.json';
+const writeFile = './data.json';
+const app = express();
+const parser = express.json();
+app.use(parser);
+
+/* An async function for grabbing the data from the json file.
+   Data will be parsed before returning.                    */
+async function getJSON(fileUrl) {
+  try {
+    const contents = await fs.readFile(fileUrl, 'utf8');
```

Comment:
What is the value in passing in the file to this function? It will always read from the same file, so you could have `getJSON` take no parameters and then you hard-code the file name here. This would make your code simpler and easier to modify.